### PR TITLE
Switch to zest releaser

### PR DESCRIPTION
**Pull request by graingert**
### Contributor Checklist:

* [x] I have updated the release notes at `docs/source/NEWS.rst`
* [ ] I have updated the automated tests.
* [ ] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by graingert** [comment]
This needs approval - as the changes are already released to pypi https://pypi.org/project/ldaptor/20.0.0rc1/

File: .github/workflows/main.yml
Comment (altendky) [inline .github/workflows/main.yml:72]:
```suggestion
    - name: Display structure of downloaded files
```

**Review by altendky — CHANGES_REQUESTED** [review]
I don't know zest.  Should I be learning it or rubber stamping to have a checkmark?

**Comment by graingert** [comment]
@altendky just a rubber stamp as this has gone out to pypi. I'll apply the changes in a follow-up

**Review by altendky — APPROVED** [review]
consider the rubber stamp applied

**Comment by graingert** [comment]
@psi29a I sent you encrypted messages on Keybase. You can install it here: https://keybase.io/phone-app

**Comment by psi29a** [comment]
Can't merge without approval... don't have access via mobile... :shrug

**Comment by psi29a** [comment]
Ha, cheated... used desktop mode.

File: CONTRIBUTING.rst
Comment (adiroiban) [inline CONTRIBUTING.rst:53]: can we run this without `pipx` ?

My vision for a new release would have been:

* Create a new PR in which the version number, release notes and release dates are manually updated
* have the PR reviewed and approved and then merge the PR
* once the PR is merged create a tag with git and push the tag, or use GitHub to create a new release page https://github.com/twisted/ldaptor/releases/new

Also, I see that in this PR the version was updated to `rc2.dev0` is there any convention for that.
Why not use just `2.0.0.dev0` or `2.0.0.rc2`

In semantic versioning I see `2.0.0-rc1` or `2.0.0-dev` ... for a pre-release https://semver.org/#spec-item-9

Comment (graingert) [reply CONTRIBUTING.rst:53]: @adiroiban this is to distinguish between releases on pypi and releases on github

**Pull request merged**
